### PR TITLE
Bump apache httpclient from 5.1 to 5.2.

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs and @jonmv PR
Some api methods are deprecated, hence the deprecation suppressions.
Merge and then rewrite ?

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java [hunk 1]
```diff
@@ -10,47 +11,75 @@
...
+    public VespaHttpClientBuilder setSocketTimeout(long connectTimeout, TimeUnit timeUnit) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout, timeUnit);
+        return this;
+    }
+    public VespaHttpClientBuilder setSocketTimeout(Timeout connectTimeout) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout);
+        return this;
+    }
+
+    public HttpClientBuilder build() {
```
Comment (bjorncs) [inline http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java:80]: Consider adding a `buildClient()`/`build()` that build the client. The `.build().build()` suffix looks a bit silly. 
Then rename previous `build()` to `builder()`/`apacheBuilder()` or something similar.

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java [hunk 2]
```diff
@@ -10,47 +11,75 @@
...
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier,
-                                           boolean rewriteHttpToHttps) {
+    public VespaHttpClientBuilder set(HostnameVerifier hostnameVerifier) {
+        this.hostnameVerifier = hostnameVerifier;
+        return this;
+    }
+    public VespaHttpClientBuilder setRewriteHttpToHttps(boolean enable) {
```
Comment (bjorncs) [inline http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java:59]: Consider simplifying method names by removing `set`-prefix.

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java [hunk 3]
```diff
@@ -10,47 +11,75 @@
...
-        return create(connectionManagerFactory, hostnameVerifier, true);
+    public VespaHttpClientBuilder set(HttpClientConnectionManagerFactory connectionManagerFactory) {
+        this.connectionManagerFactory = connectionManagerFactory;
+        return this;
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier,
-                                           boolean rewriteHttpToHttps) {
+    public VespaHttpClientBuilder set(HostnameVerifier hostnameVerifier) {
```
Comment (bjorncs) [inline http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java:55]: Consider renaming to `hostnameVerifier()`

**Review by bjorncs — APPROVED** [review]
Consider suggestions from previous review

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java (hunk 3 above)
Comment (baldersheim) [reply http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java:55]: Fixed

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java (hunk 2 above)
Comment (baldersheim) [reply http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java:59]: fixed

File: http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java (hunk 1 above)
Comment (baldersheim) [reply http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java]: Fixed

**Review by bjorncs — APPROVED** [review]

**Review by bjorncs — APPROVED** [review]

**Review by bjorncs — APPROVED** [review]

**Pull request merged**